### PR TITLE
Make name field optional in InviteRecommendation

### PR DESCRIPTION
**Pull request by leogrim**
@yipingliao @2is10

### CODE REVIEW

File: kifi-backend/modules/abook/app/com/keepit/abook/commanders/ABookRecommendationCommander.scala [hunk 1]
```diff
@@ -258,15 +258,8 @@ class ABookRecommendationCommander @Inject() (
...
-        val contacts = relevantEmailAccounts(emailAccountId)
-        val (emailAddress, name) = contacts.find(_.name.isDefined) match {
-          case Some(preferredContact) => (preferredContact.email, preferredContact.name.get)
-          case _ => {
-            val emailAddress = contacts.head.email
-            (emailAddress, emailAddress.address)
-          }
-        }
-        InviteRecommendation(SocialNetworks.EMAIL, Left(emailAddress), name, None, lastInvitedAt, score)
+        val preferredContact = relevantEmailAccounts(emailAccountId).maxBy(_.name.map(_.length) getOrElse 0)
```
Comment (2is10) [inline kifi-backend/modules/abook/app/com/keepit/abook/commanders/ABookRecommendationCommander.scala:261]: I’m not sure we want to prefer contacts with the longest names. Seems like that could introduce some bias, perhaps even toward non-human contacts. Maybe I’m misreading.

**Comment by 2is10** [comment]
So all clients are already prepared to handle no name?

**Comment by 2is10** [comment]
LGTM – just the two minor questions

File: kifi-backend/modules/abook/app/com/keepit/abook/commanders/ABookRecommendationCommander.scala (hunk 1 above)
Comment (leogrim) [inline kifi-backend/modules/abook/app/com/keepit/abook/commanders/ABookRecommendationCommander.scala:261]: This is in the situation where a user has several contacts coming from different address books for the same email address (unique emailAccountId). We pick the entry with the longest name to favor a full name over just a first or last name. This is just for display, it has no impact on ranking.

Comment (2is10) [inline kifi-backend/modules/abook/app/com/keepit/abook/commanders/ABookRecommendationCommander.scala:261]: Sounds good. Thanks for explaining.

**Comment by leogrim** [comment]
The site is the only client getting these at the moment, and it supports it. After chatting with @davoda and looking at the backend, my understanding is that mobile doesn't use WTI at all but relies on some code in typeahead to get connections in some deterministic order (not by relevance).

**Pull request merged**